### ISSUE TITLE
Something like disallowPaddingNewlinesInObjectLiterals ?

### ISSUE THREAD

**Issue by angelsanz** [jscs-dev/node-jscs#1478]
I've been unable to, well, disallow padding newlines in object literals. This is a particular case of [disallowPaddingNewlinesInBlocks](https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/disallow-padding-newlines-in-blocks.js).

Can this be done with jscs as it stands today? 

In case it doesn't, I believe its implementation is trivial from [disallowPaddingNewlinesInBlocks](https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/disallow-padding-newlines-in-blocks.js), one just needs to replace `'BlockStatement'` with `'ObjectExpression'` in [line 53](https://github.com/jscs-dev/node-jscs/blob/c668ff1be0b1cfa058866048944bad3df7ff7cbc/lib/rules/disallow-padding-newlines-in-blocks.js#L53) and adjust the error messages. Another way to achieve this would be to pass a configuration to [disallowPaddingNewlinesInBlocks](https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/disallow-padding-newlines-in-blocks.js), although that might require a little more work. I mean something like

```json
"disallowPaddingNewlinesInBlocks": {
  "onlyInObjectLiterals": true
}
```

What do you think? I can submit a PR if you think this is nice to have.

**Comment by hzoo** [comment]
Since an object literal isn't a block I guess this should be a separate rule?

**Comment by angelsanz** [comment]
Yes, that sounds more reasonable, for correctness. 

Submitting in a little.

**Comment by angelsanz** [comment]
I just realised we already have [disallowPaddingNewLinesInObjects](http://jscs.info/rule/disallowPaddingNewLinesInObjects.html). 

Hence I think the matter of whether adding a new rule or a particular configuration of [disallowPaddingNewLinesInObjects](http://jscs.info/rule/disallowPaddingNewLinesInObjects.html) is still open, and this time the latter seems more reasonable to me.

**Comment by hzoo** [comment]
If we already have a similar rule then a new rule option is definitely what we want.

**Comment by angelsanz** [comment]
I believe #1487 should be clarified before we can base the implementation of this out of [disallowPaddingNewLinesInObjects](http://jscs.info/rule/disallowPaddingNewLinesInObjects.html).

**Comment by angelsanz** [comment]
Ok, so given that [disallowPaddingNewLinesInObjects](http://jscs.info/rule/disallowPaddingNewLinesInObjects.html) is taken to mean "only disallow those newlines adjacent to the curly braces in object literals" as per the resolution of #1487, I guess the sensible choice for this configuration would be that this rule be configured with a number indicating how many such newlines are allowed (the currently implemented behaviour would be 0, and I believe a popular choice would be 1).

What do you think of this idea?

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
